tooling/minimize/tests/pass/tree_borrows/cell_inside_slice_lazy_write_to_surrounding.rs

```diff
@@ -0,0 +1,31 @@
+//@ compile-flags: --minimize-tree-borrows
+
+use std::cell::Cell;
+use std::mem;
+
+struct Example {
+    outside_array: i32,
+    array: [(Cell<i32>, i32); 2]
+}
+
+fn main() {
+    // Stored as [Cell::new(0), 1, Cell::new(2), 3, Cell::new(42)] in memory.
+    let mut root = Example { outside_array: 42, array: [(Cell::new(0), 1), (Cell::new(2), 3)] };
+    assert!(mem::offset_of!(Example, outside_array) == 16);
+    assert!(mem::offset_of!(Example, array) == 0);
+
+    unsafe {
+        let ptr = &mut root.outside_array as *const i32 as *mut i32;
+        let ptr = ptr.offset(-2) as *mut (Cell<i32>, i32);
+        let ptr = std::ptr::slice_from_raw_parts(ptr, 1);
+
+        // Looks like [(Cell::new(2), 3)].
+        let slice = &*ptr;
+        let slice_ptr = slice.as_ptr() as *const i32 as *mut i32;
+
+        *slice_ptr.offset(0) = 100;
+        *slice_ptr.offset(-1) = 100;
```

Comment:
Doesn't this rely on the field order in `Example`? You need to make that type `#[repr(C)]` then.